tests/component/db/redshift/test_records_copy.py

```diff
@@ -0,0 +1,163 @@
+            }
+            records_format =\
+                DelimitedRecordsFormat(variant='bluelabs',
+                                       hints=hints)
+            unhandled_hints = set(records_format.hints.keys())
+            out = redshift_copy_options(unhandled_hints,
+                                        records_format,
+                                        fail_if_cant_handle_hint=True,
+                                        fail_if_row_invalid=True,
+                                        max_failure_rows=0)
+            self.assertEquals(out['time_format'], expectations[datetimeformattz])
+
+    def test_redshift_copy_options_datetimeformat(self):
+        # Redshift's time_format doesn't support separate
+        # configuration for datetimeformat vs datetimeformattz, but
+        # the 'auto' flag seems to work with specific things (see
+        # tests run in records_copy.py).
+        #
+        # Please verify new formats have a test run
+        # and documented in records_copy.py before putting an entry in
+        # here.
+        for datetimeformat in DATETIMEFORMAT_CASES:
+            hints = {
+                'datetimeformattz': f"{datetimeformat}OF",
+                'datetimeformat': datetimeformat,
+            }
+            records_format =\
+                DelimitedRecordsFormat(variant='bluelabs',
+                                       hints=hints)
+            unhandled_hints = set(records_format.hints.keys())
+            out = redshift_copy_options(unhandled_hints,
+                                        records_format,
+                                        fail_if_cant_handle_hint=True,
+                                        fail_if_row_invalid=True,
+                                        max_failure_rows=0)
+            self.assertEqual(out['time_format'], 'auto')
+        for datetimeformat in DATETIMEFORMAT_CASES:
+            hints = {
+                'datetimeformattz': datetimeformat,
+                'datetimeformat': datetimeformat,
+            }
+            records_format =\
+                DelimitedRecordsFormat(variant='bluelabs',
+                                       hints=hints)
+            unhandled_hints = set(records_format.hints.keys())
+            out = redshift_copy_options(unhandled_hints,
+                                        records_format,
+                                        fail_if_cant_handle_hint=True,
+                                        fail_if_row_invalid=True,
+                                        max_failure_rows=0)
+            self.assertEqual(out['time_format'], datetimeformat)
+
+    def test_redshift_copy_options_timeonlyformat(self):
+        # Redshift didn't used to have a time only format, so Records
+        # Mover doesn't yet support anything here and just treats
+        # these as strings.  Nothing should raise.
+        #
+        # TODO: Point to a backlog item
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker